addons/sale_project/views/project_task_views.xml

```diff
@@ -30,16 +30,10 @@
         <field name="arch" type="xml">
             <xpath expr="//header" position="inside">
                 <field name="has_any_so_to_invoice" invisible="1"/>
-                <button name="action_create_invoice" string="Create Invoice" type="object" class="oe_highlight oe_read_only" groups="sales_team.group_sale_salesman_all_leads" attrs="{'invisible': [('has_any_so_to_invoice', '=', False)]}"/>
-            </xpath>
-            <xpath expr="//button[@name='%(project.act_project_project_2_project_task_all)d']" position="before">
-                <button class="d-none d-md-inline oe_stat_button"
-                        type="object" name="action_view_so" icon="fa-dollar"
-                        attrs="{'invisible': [('sale_order_id', '=', False)]}"
-                        string="Sales Order"
-                        groups="sales_team.group_sale_salesman_all_leads">
-                    <field name="sale_order_id" attrs="{'invisible': True}"/> 
-                </button>
+                <field name="display_create_invoice_primary" invisible="1"/>
+                <field name="display_create_invoice_secondary" invisible="1"/>
```

Comment:
Since the code in the compute method of those fields are the same (same logic). We could have 2 fields rather than three. For instance:
1. has_so_to_invoice: this field will be `True` if the SO in the project can be invoiced.
2. has_any_tasks_so_to_invoice: this field will be `True` if at least one SO in tasks related to the project can be invoiced.

What do you think ? 🙂

Reply:
That's not exactly why I introduced these two fields: I didn't do it to separate projects from tasks which have an invoice status == "to invoice", but as a way to distinguish if the project or its tasks have a status invoice set a "to invoice" or "no". But I agree, maybe there is a way to do it with only two fields as you suggest.